src/components/toast/global_toast_list.test.tsx

```diff
@@ -186,5 +187,100 @@ describe('EuiGlobalToastList', () => {
         expect(dismissToastSpy).toBeCalled();
       });
     });
+
+    describe('clear all toasts button', () => {
+      const createMockToasts = (toastCount: number) =>
+        Array.from(new Array(toastCount)).map((_, idx) => ({
+          id: String(idx),
+          'data-test-subj': String(idx),
+        }));
+
+      const sharedProps = {
+        toastLifeTimeMs: 10,
+        dismissToast: jest.fn(),
+        toasts: createMockToasts(CLEAR_ALL_TOASTS_THRESHOLD_DEFAULT),
+      };
+
+      afterEach(() => {
+        sharedProps.dismissToast.mockClear();
+      });
+
+      test('is not visible when the showClearAllButtonAt prop is set to 0', () => {
+        const dismissAllToastsSpy = jest.fn();
+
+        const { queryByTestSubject } = render(
+          <EuiGlobalToastList
+            {...sharedProps}
+            onClearAllToasts={dismissAllToastsSpy}
+            showClearAllButtonAt={0}
+          />
+        );
+
+        expect(
+          queryByTestSubject('euiClearAllToastsButton')
+        ).not.toBeInTheDocument();
+      });
+
+      it('is not visible when the number of toasts to be displayed is below the set toast dismiss threshold', () => {
+        const TOAST_COUNT = CLEAR_ALL_TOASTS_THRESHOLD_DEFAULT - 1;
+
+        const { queryByTestSubject } = render(
+          <EuiGlobalToastList
+            {...sharedProps}
+            toasts={createMockToasts(TOAST_COUNT)}
+          />
+        );
+
+        expect(
+          queryByTestSubject('euiClearAllToastsButton')
+        ).not.toBeInTheDocument();
+      });
+
+      test('is displayed when the number of toasts to be displayed equals the internal default threshold value', () => {
+        const dismissAllToastsSpy = jest.fn();
+
+        const { getByTestSubject } = render(
+          <EuiGlobalToastList
+            {...sharedProps}
+            onClearAllToasts={dismissAllToastsSpy}
+          />
```

Comment:
Just noticed this very quick cleanup item - the `onClearAllToasts` prop isn't being used in this test, so we should go ahead and remove it. I'll commit this suggest after and auto merge this PR for ya :)
```suggestion
        const { getByTestSubject } = render(
          <EuiGlobalToastList {...sharedProps} />
        );
```